lib.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2016 Vulcand Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicate
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// GetStringMapValue is a helper function that returns property
+// from map[string]string or map[string][]string
+// GetMapValure returns empty value in case if key not found
+// In case if map is nil, returns empty value as well
+func GetStringMapValue(mapVal, keyVal interface{}) (interface{}, error) {
+	key, ok := keyVal.(string)
+	if !ok {
+		return nil, trace.BadParameter("")
+	}
+	switch m := mapVal.(type) {
+	case map[string][]string:
+		if len(m) == 0 {
+			var n []string
```

Comment:
You could just do `return []string{}, nil` so you don't need the var?

Reply:
to return nil of a proper type